lib/experimental/class-wp-webfonts.php

```diff
@@ -268,150 +362,439 @@ public function validate_webfont( $webfont ) {
-			 * and then getting the CSS from the provider.
-			 */
-			$provider = new $provider_class();
-			$provider->set_webfonts( $provider_webfonts );
-			$styles .= $provider->get_css();
+			$this->remove_from_to_do_queues( $handle );
 		}
+	}
 
-		return $styles;
+	/**
+	 * Removes the handle from the `to_do` and `to_do_keyed_handles` lists.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param string $handle Handle to remove.
+	 */
+	private function remove_from_to_do_queues( $handle ) {
+		unset(
+			$this->to_do[ $this->to_do_keyed_handles[ $handle ] ],
+			$this->to_do_keyed_handles[ $handle ]
+		);
 	}
 
+	/**
+	 * Sets the given handle to done by adding it to the `done` list.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param string $handle Handle to set as done.
+	 */
+	private function set_as_done( $handle ) {
+		if ( ! is_array( $this->done ) ) {
+			$this->done = array();
+		}
+		$this->done[] = $handle;
+	}
 
 	/**
-	 * Reorganizes webfonts grouped by font-family into grouped by provider.
+	 * Gets the font slug.
 	 *
-	 * @param array[] $font_families Font families and each of their webfonts.
-	 * @return array[] Webfonts organized by providers.
+	 * BACKPORT NOTE: Do not backport this method.
+	 *
+	 * @since X.X.X
+	 * @deprecated Use WP_Webfonts_Utils::convert_font_family_into_handle() or WP_Webfonts_Utils::get_font_family_from_variation().
+	 *
+	 * @param array|string $to_convert The value to convert into a slug. Expected as the web font's array
+	 *                                 or a font-family as a string.
+	 * @return string|false The font slug on success, or false if the font-family cannot be determined.
 	 */
-	private function get_webfonts_by_provider( array $font_families ) {
-		$providers            = $this->get_providers();
-		$webfonts_by_provider = array();
+	public static function get_font_slug( $to_convert ) {
+		_deprecated_function(
+			__METHOD__,
+			'6.1.0',
+			'Use WP_Webfonts_Utils::get_font_family_from_variation() to get the font family from an array and WP_Webfonts_Utils::convert_font_family_into_handle() to get the handle'
+		);
 
-		foreach ( $font_families as $webfonts ) {
-			foreach ( $webfonts as $webfont ) {
-				$provider = $webfont['provider'];
+		if ( is_array( $to_convert ) ) {
+			if ( isset( $to_convert['font-family'] ) ) {
+				$to_convert = $to_convert['font-family'];
+			} elseif ( isset( $to_convert['fontFamily'] ) ) {
+				$to_convert = $to_convert['fontFamily'];
+			} else {
+				_doing_it_wrong( __METHOD__, __( 'Could not determine the font family name.', 'gutenberg' ), '6.0.0' );
+				return false;
+			}
+		}
 
-				// Skip if the provider is not registered.
-				if ( ! isset( $providers[ $provider ] ) ) {
-					/* translators: %s is the provider name. */
-					trigger_error( sprintf( __( 'Webfont provider "%s" is not registered.', 'gutenberg' ), $provider ) );
-					continue;
-				}
+		return sanitize_title( $to_convert );
+	}
```

Comment:
Can probably be simplified:

```php
		if ( is_array( $to_convert ) ) {
			if ( isset( $to_convert['font-family'] ) ) {
				return sanitize_title( $to_convert['font-family'] );
			}
			if ( isset( $to_convert['fontFamily'] ) ) {
				return sanitize_title( $to_convert['fontFamily'] );
			}
		}
		_doing_it_wrong( __METHOD__, __( 'Could not determine the font family name.', 'gutenberg' ), '6.0.0' );
		return false;
```

Reply:
It could be. But it's a copy of the code from the current API. As this method is deprecated to be removed, I was thinking of keeping untouched. What do you think @aristath?

Reply:
I suppose if it's going to be removed and not merged in Core, it's fine 👍